src/hono.ts

```diff
@@ -209,6 +228,29 @@ export const app = new Hono()
     }
   )
 
+  // Get Exchange State
+  .get(
+    routes.exchangeDetail,
+    authenticateTenantMiddleware,
+    addWorkflowByParam,
+    async (c) => {
+      const exchange = await getExchangeData(
+        c.req.param('exchangeId')!,
+        c.var.workflow.id
+      )
+      const authEnabled = c.var.config.tenantAuthenticationEnabled
+      if (
+        authEnabled &&
+        c.var.authTenant &&
+        c.var.authTenant.tenantName !== exchange?.tenantName
+      ) {
+        throw new HTTPException(401, { message: 'Unauthorized' })
+      }
```

Comment:
Some DRYness?

```suggestion
      checkAuth(c)
```

```ts
function checkAuth(c) {
      const authEnabled = c.var.config.tenantAuthenticationEnabled
      if (
        authEnabled &&
        c.var.authTenant &&
        c.var.authTenant.tenantName !== exchange?.tenantName
      ) {
        throw new HTTPException(401, { message: 'Unauthorized' })
      }
}
```

Reply:
Yeah, it does seem appealing. The reason I didn't quite do it was that the type hints are valuable about what properties `c` the context object has, and I wasn't quite sure the most useful way to express the "Context with authTenant var" if this was a reusable function, though there are some opportunities for this to be cleaner.

Also, the two different (legacy DCC, VC-API 0.7) endpoints have different request body structures so in one place it's `requestBody.tenantName` and in the other it's `requestBody.variables.tenantName`.